### ISSUE TITLE
Upcoming v0.63.3 cherrypicks discussion

### ISSUE THREAD

**Issue by grabbou** [react-native-community/releases#203]
Conversations on this thread are limited to 0.63 releases major issues and backport (cherry-pick) requests from commits that are already on master.

An example of a good such request is a bug fix for a serious issue that has been merged into master but did not make the 0.63.2 cut.

In other words, if you cannot point to a particular commit on master, then your request likely belongs as a new issue in http://github.com/facebook/react-native/issues.

**Comment by grabbou** [comment]
(I am always scared I will skip one minor version)

**Comment by nhuson269** [comment]
My application needs to solve this problem, I have had an issue with iOS since upgrading to version 0.63.x

https://github.com/facebook/react-native/issues/29021

**Comment by 1awaleed** [comment]
> My application needs to solve this problem, I have had an issue with iOS since upgrading to version 0.63.x
> 
> [facebook/react-native#29021](https://github.com/facebook/react-native/issues/29021)

This issue is breaking our apps.

**Comment by kelset** [comment]
@nhuson269 @1awaleed this issue is only for commits to cherry pick

**Comment by daveols** [comment]
@grabbou just a heads up that there's a typo the title assuming this is the thread for v0.63.**3**

**Comment by daveols** [comment]
@grabbou just a heads up that there's a typo in the title assuming this is the thread for v0.63.**3**

**Comment by sunnylqm** [comment]
The title is always confusing. Can we name it more clear like "V0.63.2(released/upcoming) Cherry Picks Discussion" @grabbou

**Comment by makarkotlov** [comment]
it seems like we need to fix this asap 2ecd79e431d16c696b486b828b322035e7ce6862

**Comment by SaeedZhiany** [comment]
> @grabbou it seems like we need to fix this asap
commit: 2ecd79e431d16c696b486b828b322035e7ce6862
pr: facebook/react-native#29477

@kelset, It seems even changing the issue title has not the proper effect 😄 

@makarkotlov in such issues, you should only request the commits to cherry-pick that have been merged on the branch master. not the open PRs.

**Comment by alloy** [comment]
@TheSavior Can we get this merged so we can pick it here? https://github.com/facebook/react-native/pull/29074

**Comment by makarkotlov** [comment]
@SaeedZhiany how silly of me(( I came from here https://github.com/facebook/react-native/releases/tag/v0.63.2 and there's a statement "You can participate in the conversation for the next patch release in the dedicated issue." and I thought that I can propose there to merge my pr quicker

**Comment by sammy-SC** [comment]
@grabbou could I get https://github.com/facebook/react-native/commit/ffc90c7f92e63e1a53ed107833e3deed492ab435 picked ? 

it resolves https://github.com/facebook/react-native/issues/29364

**Comment by cristianoccazinsp** [comment]
This issue https://github.com/facebook/react-native/issues/29451 needs some urgent attention.  RN 0.63 is pretty much unusable on android when the bug happens.

**Comment by trcoffman** [comment]
Please get both of these released in tandem!

https://github.com/facebook/react-native/commit/e566c7ec7b0015e4e5e6ad92247a6357eb861d4f
https://github.com/facebook/hermes/commit/d188b722bae7664db4a14898c9fde189353034ef

**Comment by caiangums** [comment]
> @grabbou could I please get [facebook/react-native@ffc90c7](https://github.com/facebook/react-native/commit/ffc90c7f92e63e1a53ed107833e3deed492ab435) picked ?
> 
> it resolves [facebook/react-native#29364](https://github.com/facebook/react-native/issues/29364)
> 
> is there any other information I need to provide?
> 
> Thank you

I think if this really solves the issue for iOS, it would be nice to have! 👍  for this one!

**Comment by alphasp** [comment]
Image api `getSizeWithHeaders` on iOS is currently broken. 
This commit fixed it: https://github.com/facebook/react-native/commit/0bcc686c1cc90fd44de7a28e2f56ea20fe2f5123

**Comment by affantechcarrot** [comment]
Hi When this will be release ?

**Comment by kelset** [comment]
friendly reminder folks: it's August. Expect things to slow down.

There are currently only 3 commits requested that are cherry pickable:
* https://github.com/facebook/react-native/commit/dc2df754267df3909631d81c22b9fcab58dfa241
* https://github.com/facebook/react-native/commit/ffc90c7f92e63e1a53ed107833e3deed492ab435
* https://github.com/facebook/react-native/commit/0bcc686c1cc90fd44de7a28e2f56ea20fe2f5123

The rest of the links are not actionables currently (one is an open PR and the other requires an Hermes releases AFAIK so not doable for a patch release).

**Comment by laurenwilliams** [comment]
Would love to see this released if possible to fix an issue with request tokens on image uploads for iOS:

https://github.com/facebook/react-native/commit/ffc90c7f92e63e1a53ed107833e3deed492ab435

**Comment by oleksandr-dziuban** [comment]
Hello guys, this PR is critical, as we can't use babel transform plugins with react-native 0.63.0+
https://github.com/facebook/react-native/pull/29477

**Comment by oleksandr-dziuban** [comment]
Additionally raised this issue:
https://github.com/facebook/react-native/issues/29636

It is critical issue for projects with this babel plugin

**Comment by oleksandr-dziuban** [comment]
Yes, I can confirm, this PR fixes the issue:
https://github.com/facebook/react-native/pull/29477

I have assembled release iOS build locally with this change and it works!

**Comment by mikehardy** [comment]
@oleksandr-dziuban we need commit hashes here, not PRs - please post the commit hashes of things to cherry-pick for release once they are merged

**Comment by lorenzoangelini** [comment]
@oleksandr-dziuban https://github.com/facebook/react-native/commit/2ecd79e431d16c696b486b828b322035e7ce6862

**Comment by kelset** [comment]
@lorenzoangelini that commit hash is not coming from master - until the PR is merged to master it can't be cherry picked.

**Comment by lorenzoangelini** [comment]
@kelset  is it possible to merge it? For a lot of developers and company this issues are blocking the upgrade to 0.63.2

**Comment by kelset** [comment]
@lorenzoangelini I can't merge it, it needs to be reviewed and tested and then imported in the FB codebase. I know that the author of the commit being reverted will be looking into this probably next week.

**Comment by Manouli** [comment]
> friendly reminder folks: it's August. Expect things to slow down.
> 
> There are currently only 3 commits requested that are cherry pickable:
> 
> * [facebook/react-native@dc2df75](https://github.com/facebook/react-native/commit/dc2df754267df3909631d81c22b9fcab58dfa241)
> * [facebook/react-native@ffc90c7](https://github.com/facebook/react-native/commit/ffc90c7f92e63e1a53ed107833e3deed492ab435)
> * [facebook/react-native@0bcc686](https://github.com/facebook/react-native/commit/0bcc686c1cc90fd44de7a28e2f56ea20fe2f5123)
> 
> The rest of the links are not actionables currently (one is an open PR and the other requires an Hermes releases AFAIK so not doable for a patch release).

Yep please 🙏 Our app is broken right now in production due the token issue it would be great if you could kickstart this. Thanks again.

**Comment by gaodeng** [comment]
Please revert this PR   https://github.com/facebook/react-native/pull/28354  which break iOS archive release process
more info https://github.com/facebook/react-native/pull/29477

**Comment by heroic** [comment]
Can we merge https://github.com/facebook/react-native/commit/07a597ad185c8c31ac38bdd4d022b0b880d02859

**Comment by inuyashaaa** [comment]
Is there any update on the version?

**Comment by SuperDuperMartin** [comment]
Would be awesome to get this cherry-picked now ⭐

**Comment by kelset** [comment]
hey folks, currently no updates. We are still waiting on https://github.com/facebook/react-native/pull/29477 to be merged since that should also be related to helping us move closer to be able to cut 0.64. What is likely to happen is 0.64rc0 first, then probably 0.63.3 (unless upgrading to 0.64 looks trivial in which case we'll focus on getting to 0.64.0 directly). Sorry for the delays, summer/end of summer are always complicated.

**Comment by yungsters** [comment]
Can we merge https://github.com/facebook/react-native/commit/86ffb9c41e033f59599e01b7ad016706b5f62fc8 for https://github.com/facebook/react-native/issues/29376?

**Comment by safaiyeh** [comment]
Can we cherrypick https://github.com/facebook/react-native/commit/d8b70b19b39ead4dd41895d666d116a43c56474e
issue: https://github.com/facebook/react-native/issues/28551

and make sure https://github.com/react-native-community/releases/issues/203#issuecomment-665754860 as well makes it in. issue: https://github.com/facebook/react-native/issues/29364

Both resolves image issues on Android & iOS that has had 100+ comments on each of their issues.

**Comment by SandroMachado** [comment]
Can we merge https://github.com/facebook/react-native/commit/45954ac5dccdfe05de7553a0f08c4f0e66e3d62e?

**Comment by alloy** [comment]
While not yet merged, I expect it to be merged soon. https://github.com/facebook/react-native/pull/29995

**Comment by cristeahub** [comment]
Can we cherry pick https://github.com/facebook/react-native/commit/123423c2a9258c9af25ca9bffe1f10c42a176bf3 ? Building with iOS 14 leaves invisible images without it.

**Comment by sunnylqm** [comment]
@cristeahub it's already been merged into 0.63.2 https://github.com/facebook/react-native/commit/b6ded7261544c703e82e8dbfa442dad4b201d428

**Comment by cristeahub** [comment]
> @cristeahub it's already been merged into 0.63.2 facebook/react-native@b6ded72

Great! Thanks for letting me know 👍  From the commit on master I couldn't see that it was a part of a release.

**Comment by pokey** [comment]
> hey folks, currently no updates. We are still waiting on [facebook/react-native#29477](https://github.com/facebook/react-native/pull/29477) to be merged since that should also be related to helping us move closer to be able to cut 0.64. What is likely to happen is 0.64rc0 first, then probably 0.63.3 (unless upgrading to 0.64 looks trivial in which case we'll focus on getting to 0.64.0 directly). Sorry for the delays, summer/end of summer are always complicated.

Hi @kelset; looks like facebook/react-native#29477 is merged.  Any updates on release?  Would be great to have that bundler working in staging and release again 😅

**Comment by alloy** [comment]
@kelset is currently on vacation, so I’ll try to find some time later this week or Monday to cut a release.

**Comment by a-eid** [comment]
Does any one having issues with core IOS `Modal` component in `0.63.2` or is it just me, 
am also wondering if has changed from `0.62.2` version ?

I've made an issue here but I'm not sure if it's always reproducable so I closed it & reverted my projects back to 0.62.2 ( mainly because of the ios release builds. ). 

https://github.com/facebook/react-native/issues/29973

**Comment by alloy** [comment]
@a-eid As per the OP, this issue is only to discuss cherry-picking fixes that have landed in the `master` branch.

**Comment by kelset** [comment]
@pokey I hope you realize you wrote your message something like 4hrs after the PR was merged and it's been merged for less than 24hrs at the time of me writing this. So @-ing me in expecting an update is quite.... I mean 😕

---

That said, let's try to sum up the current status of things:

Commits requested related to fixes:
* https://github.com/facebook/react-native/commit/83777cb4fb5dda89c430b7eff9cd1f28d2577831
* https://github.com/facebook/react-native/commit/dc2df754267df3909631d81c22b9fcab58dfa241
* https://github.com/facebook/react-native/commit/ffc90c7f92e63e1a53ed107833e3deed492ab435
* https://github.com/facebook/react-native/commit/0bcc686c1cc90fd44de7a28e2f56ea20fe2f5123
* https://github.com/facebook/react-native/commit/07a597ad185c8c31ac38bdd4d022b0b880d02859
* https://github.com/facebook/react-native/commit/86ffb9c41e033f59599e01b7ad016706b5f62fc8
* https://github.com/facebook/react-native/commit/45954ac5dccdfe05de7553a0f08c4f0e66e3d62e

This commit https://github.com/facebook/react-native/commit/d8b70b19b39ead4dd41895d666d116a43c56474e has been requested but since it changes the version of Flipper, I have reservations of making this happen in a patch release.

Aside from the above, there are 2 more things to keep an eye on:
* https://github.com/facebook/react-native/pull/29995 requested by @alloy not yet merged
* since XCode12 has been released, there has been some issues https://github.com/facebook/react-native/issues/29984 and while @grabbou has provided a temp workaround it might be necessary for a PR to come up to fix it properly. But at this point in time there's not enough info around it - so it's more likely that worst case it's going to be a discussion for 0.63.4.

**Comment by a-eid** [comment]
this fixes a critical issue with IOS modal in dev mode 

https://github.com/facebook/react-native/pull/29787

**Comment by pokey** [comment]
@kelset thanks for the update!  didn't realize facebook/react-native#29477 was merged so shortly before my comment 😅.  I stumbled across that PR while in the midst of a furious debugging session , then saw this thread, and figured I'd bump, without seeing the timeline.

Thanks again for your responsiveness!

**Comment by jinshin1013** [comment]
Is it too late to add a commit for this version?

Fixes the iOS Gif not animating when TurboModule is enabled: https://github.com/facebook/react-native/commit/663b5a878be9faafd13b41c222a1bc2ac7bb3a65.

**Comment by RigoTheDev** [comment]
Don't know why there isn't much of an urgency to get these fixes out as they are pretty major :/

**Comment by piashcse** [comment]
is there any plan of releasing 0.64 rc0 ?

**Comment by kelset** [comment]
@piashcse  there's a dedicated issue for that #207

**Comment by kelset** [comment]
Here's the list of commits we'll attempt to cherry pick for 0.63.3, in chrono order (oldest -> newest):

* https://github.com/facebook/react-native/commit/45954ac5dccdfe05de7553a0f08c4f0e66e3d62e
* https://github.com/facebook/react-native/commit/ffc90c7f92e63e1a53ed107833e3deed492ab435
* https://github.com/facebook/react-native/commit/0bcc686c1cc90fd44de7a28e2f56ea20fe2f5123
* https://github.com/facebook/react-native/commit/dc2df754267df3909631d81c22b9fcab58dfa241
* https://github.com/facebook/react-native/commit/07a597ad185c8c31ac38bdd4d022b0b880d02859
* https://github.com/facebook/react-native/commit/86ffb9c41e033f59599e01b7ad016706b5f62fc8
* https://github.com/facebook/react-native/commit/d8b70b19b39ead4dd41895d666d116a43c56474e
* https://github.com/facebook/react-native/commit/83777cb4fb5dda89c430b7eff9cd1f28d2577831
* https://github.com/facebook/react-native/commit/6e08f84719c47985e80123c72686d7a1c89b72ed

In the changelog we should also still mention/redirect people to this issue https://github.com/facebook/react-native/issues/29984 in case they have problems with Xcode12

**Comment by ravirajn22** [comment]
> @grabbou could I please get [facebook/react-native@ffc90c7](https://github.com/facebook/react-native/commit/ffc90c7f92e63e1a53ed107833e3deed492ab435) picked ?
> 
> it resolves [facebook/react-native#29364](https://github.com/facebook/react-native/issues/29364)
> 
> is there any other information I need to provide?
> 
> Thank you

Request RN team to at least pick this commit and release a version. It's a real deal breaker to upgrade. Sorry if I am sounding harsh, but it is not fair to take more than 2 months to release a vital bug fix in fact it was a regression.

I think there is nothing wrong in having many patch releases, instead of waiting for one big patch release which has caused this delay.

I would be happy to help in my capacity, thanks.

**Comment by diegolmello** [comment]
> > @grabbou could I please get [facebook/react-native@ffc90c7](https://github.com/facebook/react-native/commit/ffc90c7f92e63e1a53ed107833e3deed492ab435) picked ?
> > it resolves [facebook/react-native#29364](https://github.com/facebook/react-native/issues/29364)
> > is there any other information I need to provide?
> > Thank you
> 
> Request RN team to at least pick this commit and release a version. It's a real deal breaker to upgrade. Sorry if I am sounding harsh, but it is not fair to take more than 2 months to release a vital bug fix in fact it was a regression.
> 
> I think there is nothing wrong in having many patch releases, instead of waiting for one big patch release which has caused this delay.
> 
> I would be happy to help in my capacity, thanks.

@ravirajn22 IMO, in cases like these, you should do a [patch-package](https://github.com/ds300/patch-package) picking the fix to your local RN. It's not hard and demands a small effort.
These are dark times to blame the community for things you could fix easily.

**Comment by ravirajn22** [comment]
@diegolmello I don't want to get into argument here, just think for a second if it valid from RN team for taking more than 2 months to fix a regression. I think if have to defend and criticise appropriately. 

Instead of asking everyone to custom patch, just a release would have improved DX greatly. Moreover I am not commenting here to point fingers or blame anyone. Having a prompt service would help bringing new developers to react-native and provide confidence to existing ones. My ultimate aim here is not my upgrade process, but overall communities.

**Comment by diegolmello** [comment]
@ravirajn22 Yep, I see your point and I don't want to argue as well (you can mark us off-topic if needed).
I just commented because `it's not fair` is a hot take.

**Comment by mikehardy** [comment]
There is an off-topic discussion here about patching RN, I just want to say that I am a huge proponent of patch-package and "do it yourself" (read as: make a patch and move on, don't depend on release cycles) but patching react-native is very difficult because it's packaged as .aar files etc, you have to do a custom build from source basically, it is much more involved.

Although perhaps I am missing how you can patch-package a react-native issue? That would be great to know and perhaps provide real value from this comment.

So there is a reasonable point to be made that faster releases would be great but it's also vital to note it's community-driven and 2020 is just not a great year.

I will now self-mark this comment as off-topic and I apologize for the noise.

Thanks for for your efforts @grabbou and @kelset

**Comment by kelset** [comment]
Status of branch after cherry picking; final list of commits cherry picked:

* https://github.com/facebook/react-native/commit/45954ac5dccdfe05de7553a0f08c4f0e66e3d62e
* https://github.com/facebook/react-native/commit/ffc90c7f92e63e1a53ed107833e3deed492ab435
	* this had a conflict but I think I resolved it properly
* https://github.com/facebook/react-native/commit/0bcc686c1cc90fd44de7a28e2f56ea20fe2f5123
* https://github.com/facebook/react-native/commit/dc2df754267df3909631d81c22b9fcab58dfa241
* https://github.com/facebook/react-native/commit/07a597ad185c8c31ac38bdd4d022b0b880d02859
* https://github.com/facebook/react-native/commit/86ffb9c41e033f59599e01b7ad016706b5f62fc8
* https://github.com/facebook/react-native/commit/d8b70b19b39ead4dd41895d666d116a43c56474e
	* conflict here too but on readme and podlock file so resolved easily
* https://github.com/facebook/react-native/commit/83777cb4fb5dda89c430b7eff9cd1f28d2577831
* https://github.com/facebook/react-native/commit/6e08f84719c47985e80123c72686d7a1c89b72ed
	* another conflict here in the podspec, I think I fixed it the right way (in this branch Folly is still used as is and not as RTC-Folly as far as I can see)
* https://github.com/facebook/react-native/commit/663b5a878be9faafd13b41c222a1bc2ac7bb3a65

We will now do some testing both locally and on CI and see if everything goes well. If that's the case, we will proceed with the release.

**Comment by brentvatne** [comment]
this may be worth cherry picking as well: https://github.com/facebook/react-native/commit/43c161c30aff2f5306d32c9994fefc9edae08f85

it fixes https://github.com/facebook/react-native/issues/29360

**Comment by kelset** [comment]
@brentvatne looks like that commit is already present in the 0.63 branch.

**Comment by kelset** [comment]
Allright folks, after all that rollercoaster we finally managed to push through and release 0.63.3!

It's been a wild ride, hopefully now we can focus on getting the final touches ready cutting the 0.64 branch - in case something wild happen, please refer to the new issue: https://github.com/react-native-community/releases/issues/212
